core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/BCFile.java

```diff
@@ -361,21 +326,8 @@ public Writer(FSDataOutputStream fout, RateLimiter writeLimiter, String compress
       fsOutputBuffer = new BytesWritable();
       Magic.write(this.out);
 
-      // Set up crypto-related detail, including secret key generation and encryption
-
-      this.cryptoModule = CryptoModuleFactory.getCryptoModule(accumuloConfiguration);
-      this.cryptoParams = new BCFileCryptoModuleParameters();
-      CryptoModuleFactory.fillParamsObjectFromConfiguration(cryptoParams, accumuloConfiguration);
-      this.cryptoParams = (BCFileCryptoModuleParameters) cryptoModule
-          .generateNewRandomSessionKey(cryptoParams);
-
-      this.secretKeyEncryptionStrategy = CryptoModuleFactory
-          .getSecretKeyEncryptionStrategy(accumuloConfiguration);
-      this.cryptoParams = (BCFileCryptoModuleParameters) secretKeyEncryptionStrategy
-          .encryptSecretKey(cryptoParams);
-
-      // secretKeyEncryptionStrategy.encryptSecretKey(cryptoParameters);
-
+      this.encryptionStrategy = EncryptionStrategyFactory
+          .setupConfiguredEncryption(accumuloConfiguration, EncryptionStrategy.Scope.RFILE);
```

Comment:
For a new pluggable interface that we expect users to implement directly as (eventually) public API, we should avoid passing in `AccumuloConfiguration`, which is an internal-only object type.